external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisBackedBlockRDD.scala

```diff
@@ -147,6 +152,14 @@ class KinesisSequenceRangeIterator(
   private var lastSeqNumber: String = null
   private var internalIterator: Iterator[Record] = null
 
+  // variable for kinesis wait time interval between next retry
+  private var kinesisWaitTimeMs = JavaUtils.timeStringAsMs(
```

Comment:
No, this value is modified after waits -
`kinesisWaitTimeMs *= 2  // if you have waited, then double wait time for next round`